apps/extension/src/background/index.ts

```diff
@@ -12,30 +13,54 @@ import {
 import { Ports, KVPrefix } from "router";
 import { defaultChainId, chains } from "@anoma/chains";
 import { ChainsService, init as initChains } from "./chains";
-import { KeyRingService, init as initKeyRing } from "./keyring";
-
-initCrypto();
-initShared();
+import { KeyRingService, init as initKeyRing, SDK_KEY } from "./keyring";
 
 const messenger = new ExtensionMessenger();
 const store = new IndexedDBKVStore(KVPrefix.IndexedDB);
-const extensionStore = new ExtensionKVStore(KVPrefix.LocalStorage, {
-  get: browser.storage.local.get,
-  set: browser.storage.local.set,
-});
-const router = new ExtensionRouter(
-  ContentScriptEnv.produceEnv,
-  messenger,
-  extensionStore
-);
-router.addGuard(ExtensionGuards.checkOriginIsValid);
-router.addGuard(ExtensionGuards.checkMessageIsInternal);
-
-const chainsService = new ChainsService(store, [chains[defaultChainId]]);
-const keyRingService = new KeyRingService(store, defaultChainId);
-
-// Initialize messages and handlers
-initChains(router, chainsService);
-initKeyRing(router, keyRingService);
-
-router.listen(Ports.Background);
+// TODO: For now we will be running two stores side by side
+const sdkStore = new IndexedDBKVStore(KVPrefix.SDK);
+
+const DEFAULT_URL =
+  "https://d3brk13lbhxfdb.cloudfront.net/qc-testnet-5.1.025a61165acd05e";
+const { REACT_APP_NAMADA_URL = DEFAULT_URL } = process.env;
+
+(async function init() {
+  await initCrypto();
+  await initShared();
+
+  const extensionStore = new ExtensionKVStore(KVPrefix.LocalStorage, {
+    get: browser.storage.local.get,
+    set: browser.storage.local.set,
+  });
+
+  const router = new ExtensionRouter(
+    ContentScriptEnv.produceEnv,
+    messenger,
+    extensionStore
+  );
+  router.addGuard(ExtensionGuards.checkOriginIsValid);
+  router.addGuard(ExtensionGuards.checkMessageIsInternal);
+
+  // TODO: Stored data is not encrypted in any way
```

Comment:
Probably because I'm passing None as password :D I think it can be fixed in the next issue.